### ISSUE TITLE
MonoApplicative

### ISSUE THREAD

**Issue by sheganinans** [snoyberg/mono-traversable#84]
Hi! Excellent library you have here, I was wondering why there was no MonoApplicative instance in the ecosystem and I was pointed from there [1] to here as the right spot to ask about this. Thanks!

[1] https://www.reddit.com/r/haskell/comments/40bpdn/endoapplicatives_anything_similar/

**Comment by snoyberg** [comment]
While I agree that you're dealing with a variety of "monomorphic container," it doesn't really fit the definition we use here in mono-traversable, as some of the commenters explained on Reddit (responding to the comment you link to).

**Comment by sheganinans** [comment]
True, although I think that there should be a single place where such a general structure could be canonical. I went ahead and made a fork and added an initial implementation of an instance.:

https://github.com/snoyberg/mono-traversable/pull/85

It works for a whole bunch of instances for free. I also was able to use it to re-implement my nestedmap instance. GHC was able to infer the Data.Map MonoApplicative instance for free, thanks to your IsMap class, I think this really proves the merit of this extension.
 
https://github.com/sheganinans/applicative-nestedmap/blob/master/src/Data/Nested/Internal.hs#L188

I was also able to update my upstream code to utilize this stack and that seems to prove it solid.
